### ISSUE TITLE
Request for Release of trained Attacker Checkpoints to Help Reproduce Results

### ISSUE THREAD

**Issue by albert-y1n** [facebookresearch/rl-injector#1]
Hello Authors,

Thank you for the excellent work! I'm trying to reproduce your results with Meta-SecAlign 8B as the target model, but I'm only achieving 56% ASR, which is significantly lower than your reported results.

Could you please consider releasing the trained attacker lora checkpoints used in your experiments? This would greatly help with reproducing your results and understanding the training process.

Thank you for your consideration!

Best regards,
Chenlong

**Comment by YuxinWenRick** [comment]
Hi Chenlong,

Thanks for reaching out! Due to company policy, we are unable to release any trained attacker checkpoints. However, I’m happy to assist you with debugging the code and reproducing the results.

If you run the script `launch_scripts/target_llama_3_8b_instruct.sh`, your training reward curve should look similar to this, and you can expect a high test ASR as well:

<img width="1060" height="598" alt="Image" src="https://github.com/user-attachments/assets/8cea1022-c731-4ec7-b1e4-9af185c42565" />

Would you be able to share your training reward plot? I’d like to take a look to see if there’s anything unusual. In the meantime, I’ll run some experiments to see if I can reproduce your issue.

Another point I’d like to mention, also mentioned [here](https://github.com/facebookresearch/rl-injector/tree/main?tab=readme-ov-file#training), is that the current library versions can sometimes be unstable. These are not the versions we used in our original experiments, but we are required to use the latest libraries for code release purposes. I haven’t explored this extensively, but it may be helpful for you to try using the original library versions. To do this, simply overwrite the `requirements.txt` file with the following:

```
vllm==0.9.2
transformers==4.52.3
datasets==3.6.0
accelerate==1.7.0
evaluate==0.4.3
bitsandbytes==0.46.0
trl==0.19.1
peft==0.15.2
pillow==11.1.0
protobuf
ipython
wandb
scikit-learn
matplotlib
litellm[proxy]
tenacity
nltk
openai==1.99.9
google-generativeai==0.8.5
anthropic==0.59.0
sentence-transformers
bert_score
```

Let me know if you have any other questions!

**Comment by albert-y1n** [comment]
Dear Yuxin, 

Thanks a lot for your timely reply and assistance. The script I ran is `launch_scripts/target_meta_secalign_8b.sh`. The only thing I changed is the batch size and the GPU num. I used 8 as the batch size and trained on 4 GPUs, but I got a training curve like this:

<img width="507" height="303" alt="Image" src="https://github.com/user-attachments/assets/a3dce203-9242-4287-80ee-65f6b25a0aa9" />

Is this a correct training curve? Thanks again for your selfless help.

Best regards,
Chenlong

**Comment by YuxinWenRick** [comment]
Hi Chenlong, thanks for sharing that information. The training curve does looks like a little bit off. Let me run the experiment with your setup (4 GPUs, bs=8) to see if I get a similar plot. If that’s the case, we might want to increase the batch size with a higher `gradient_accumulation_steps`.

**Comment by albert-y1n** [comment]
Dear Yuxin, I will try `gradient_accumulation_steps` as 4. Thanks a lot for your help!

**Comment by YuxinWenRick** [comment]
Hi Chenlong, I ran the experiments on 4 GPUs using the following hyperparameters:

```
    --num_generations 16 \
    --num_iterations 1 \
    --per_device_train_batch_size 8 \
    --gradient_accumulation_steps 2 \
```

The plot I obtained is shown here, and it seems that I can achieve high rewards in both the old and new environments:

<img width="1093" height="583" alt="Image" src="https://github.com/user-attachments/assets/2020de3c-9d01-4cbc-a2b6-a17bf719af69" />

Meanwhile, I found that training with the original environment is more stable and converges faster, likely due to some changes in the GRPO trainer implementation.

I’ve updated the instructions [here](https://github.com/facebookresearch/rl-injector?tab=readme-ov-file#original-environment) to help you recreate the original environment for our experiments. If you still encounter the same issue as before, you might want to try installing the original environment.

Please let me know if you have any other questions!

**Comment by PramuPerera** [comment]
Are these training curves still valid for commercial models?
